### PR TITLE
Solenoid states

### PR DESCRIPTION
**Pull request by NoamRosenberg08**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/solenoid/SolenoidStateHandler.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+	public SolenoidStateHandler(Solenoid solenoid) {
+		this.solenoid = solenoid;
+	}
+
+	public Command setState(SolenoidState state) {
+		return solenoid.getCommandsBuilder().setPower(state.getAppliedPower());
+	}
+
+
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/solenoid/SolenoidStateHandler.java]: remove space

File: src/main/java/frc/robot/subsystems/solenoid/SolenoidState.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package frc.robot.subsystems.solenoid;
+
+public enum SolenoidState {
+
+	OFF(0),
+	RETRACT(0.8),
+	HOLD(0.2);
+
+
+	private final double appliedPower;
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/solenoid/SolenoidState.java]:
```suggestion
	private final double power;
```

Comment (YHerm) [reply src/main/java/frc/robot/subsystems/solenoid/SolenoidState.java]: same in all  places

**Review by YHerm — COMMENTED** [review]
merge core

**Review by drorta — DISMISSED** [review]
lots of conflicts

File: src/main/java/frc/robot/subsystems/lifter/LifterStuff.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package frc.robot.subsystems.lifter;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.hardware.motor.ControllableMotor;
+import frc.robot.hardware.signal.InputSignal;
+
+public record LifterStuff(
```
Comment (tnrz135) [inline src/main/java/frc/robot/subsystems/lifter/LifterStuff.java]: Webster's dictionary states that stuff is: "materials, supplies, or equipment used in various activities"

File: src/main/java/frc/robot/subsystems/lifter/factory/LifterFactory.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import frc.robot.Robot;
+import frc.robot.subsystems.lifter.LifterStuff;
+
+public class LifterFactory {
+
+	public static LifterStuff create(String logPath) {
+		return switch (Robot.ROBOT_TYPE) {
+			case REAL -> LifterRealConstants.generateLifterStuff(logPath);
+			case SIMULATION -> null;
```
Comment (tnrz135) [inline src/main/java/frc/robot/subsystems/lifter/factory/LifterFactory.java]: Why no simulation?

File: src/main/java/frc/robot/subsystems/lifter/factory/LifterRealConstants.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import frc.robot.hardware.motor.phoenix6.TalonFXWrapper;
+import frc.robot.hardware.signal.phoenix.Phoenix6SignalBuilder;
+import frc.robot.subsystems.lifter.LifterStuff;
+import frc.utils.AngleUnit;
+
+import static edu.wpi.first.math.util.Units.inchesToMeters;
+import static edu.wpi.first.units.Units.Seconds;
+import static edu.wpi.first.units.Units.Volts;
+
+public class LifterRealConstants {
```
Comment (tnrz135) [inline src/main/java/frc/robot/subsystems/lifter/factory/LifterRealConstants.java]: Magic in this class

**Review by tnrz135 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/lifter/factory/LifterFactory.java (hunk 4 above)
Comment (NoamRosenberg08) [reply src/main/java/frc/robot/subsystems/lifter/factory/LifterFactory.java]: cuz no simulation motor, we run only on robot.

**Review by YHerm — CHANGES_REQUESTED** [review]
the solenoid states classes look good, but merge lifter here and core to lifter

**Review by YHerm — DISMISSED** [review]
merge lifter first

**Pull request merged**